src/flashcards/Main.java

```diff
@@ -1,7 +1,200 @@
 package flashcards;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Scanner;
+
 public class Main {
-    public static void main(String[] args) {
-        System.out.print("Hello world!");
+
+  private static Map<String, String> cardToDefinition = new LinkedHashMap<>();
+  private static Map<String, String> definitionToCard = new LinkedHashMap<>();
+  private static List<String> logs = new ArrayList<>();
+  private static Map<String, Integer> stats = new HashMap<>();
+
+  public static void main(String[] args) {
+    try (Scanner scanner = new Scanner(System.in)) {
+      while (true) {
+        System.out.println(
+            "Input the action (add, remove, import, export, ask, exit, log, hardest card, reset stats):");
+
+        String action = scanner.nextLine();
+
+        logs.add(action);
+
+        switch (action) {
```

Comment:
It may be useful to apply something like the command pattern together with a simple factory here instead of this big switch. It would isolate this code from possible changes in further. Each private method here can be represented like a small object which does an action.

Reply:
Thanks for review. Committed changes about that.

Reply:
No problems. In general, your code looks pretty well. 
Therefore, nothing to complain about here. You can use more advanced concepts, even if they are not yet explained in the topics. For example - Stream API and functional interfaces.